pkg/latte/intent_verify_login_link.go

```diff
@@ -0,0 +1,79 @@
+package latte
+
+import (
+	"context"
+
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/lib/workflow"
+	"github.com/authgear/authgear-server/pkg/util/validation"
+)
+
+func init() {
+	workflow.RegisterPublicIntent(&IntentVerifyLoginLink{})
+}
+
+var IntentVerifyLoginLinkSchema = validation.NewSimpleSchema(`
+	{
+		"type": "object",
+		"additionalProperties": false,
+		"properties": {
+			"code": { "type": "string" }
+		},
+		"required": ["code"]
+	}
+`)
+
+type IntentVerifyLoginLink struct {
+	Code string `json:"code"`
+}
+
+func (i *IntentVerifyLoginLink) Kind() string {
+	return "latte.IntentVerifyLoginLink"
+}
+
+func (i *IntentVerifyLoginLink) JSONSchema() *validation.SimpleSchema {
+	return IntentVerifyLoginLinkSchema
+}
+
+func (i *IntentVerifyLoginLink) GetCode() string {
+	return i.Code
+}
+
+func (i *IntentVerifyLoginLink) CanReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow) ([]workflow.Input, error) {
+	return []workflow.Input{}, nil
+}
+
+func (i *IntentVerifyLoginLink) ReactTo(ctx context.Context, deps *workflow.Dependencies, w *workflow.Workflow, input workflow.Input) (*workflow.Node, error) {
+	codeModel, err := deps.OTPCodeService.SetUserInputtedMagicLinkCode(i.GetCode())
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = deps.OTPCodeService.VerifyMagicLinkCode(i.GetCode(), false)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the web session and trigger the refresh event
+	webSessionProvider := deps.GlobalSessionServiceFactory.NewGlobalSessionService(
+		config.AppID(codeModel.AppID),
+	)
+	webSession, err := webSessionProvider.GetSession(codeModel.WebSessionID)
+	if err != nil {
+		return nil, err
+	}
+	err = webSessionProvider.UpdateSession(webSession)
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, workflow.ErrEOF
```

Comment:
@louischan-oursky Currently it returns internal error. From https://github.com/authgear/authgear-server/blob/2023C01/docs/specs/workflow.md#canreactto-and-reactto this return should return 200 with no next node?

Reply:
I also encountered a similar error, but the symptom is a finished workflow would return `continue` action. I think we don't expect `workflow.ErrEOF` to be returned by `ReactTo`, only `CanReactTo` should return this error.

Reply:
I think from spec returning `nil, workflow.ErrEOF` will exit the intent without doing anything though. 

I did check `FindInputReactor()` method and from comments `workflow.ErrEOF` is an accepted output from `ReactTo`.

Reply:
ErrEOF can only be returned by CanReactTo.